tests/test_maf.py

```diff
@@ -241,47 +241,110 @@ def test_error__check_allele_col():
+                dict(
+                    REFERENCE_ALLELE=["C", "C", "C"],
+                    TUMOR_SEQ_ALLELE1=["C", "C", "C"],
+                    TUMOR_SEQ_ALLELE2=["A", "A", "A"],
+                )
+            ),
+            "",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    REFERENCE_ALLELE=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE1=["B", "B", "B"],
+                    TUMOR_SEQ_ALLELE2=["C", "C", "C"],
+                )
+            ),
+            "maf: Contains both "
+            "TUMOR_SEQ_ALLELE1 and TUMOR_SEQ_ALLELE2 columns. "
+            "All values in TUMOR_SEQ_ALLELE1 must match all values in "
+            "REFERENCE_ALLELE or all values in TUMOR_SEQ_ALLELE2.\n",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    REFERENCE_ALLELE=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE1=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE2=["A", "C", "C"],
+                )
+            ),
+            "maf: Contains instances where values in REFERENCE_ALLELE match values in TUMOR_SEQ_ALLELE2. "
+            "This is invalid. Please correct.\n",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    REFERENCE_ALLELE=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE2=["A", "C", "C"],
+                )
+            ),
+            "maf: Contains instances where values in REFERENCE_ALLELE match values in TUMOR_SEQ_ALLELE2. "
+            "This is invalid. Please correct.\n",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    REFERENCE_ALLELE=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE2=["C", "C", "C"],
+                )
+            ),
+            "",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    TUMOR_SEQ_ALLELE1=["C", "C", "C"],
+                )
+            ),
+            "",
+        ),
+        (
+            pd.DataFrame(
+                dict(
+                    REFERENCE_ALLELE=["A", "A", "A"],
+                    TUMOR_SEQ_ALLELE1=["B", "B", "B"],
+                    TUMOR_SEQ_ALLELE2=["A", "C", "C"],
+                )
+            ),
+            "maf: Contains both "
+            "TUMOR_SEQ_ALLELE1 and TUMOR_SEQ_ALLELE2 columns. "
+            "All values in TUMOR_SEQ_ALLELE1 must match all values in "
+            "REFERENCE_ALLELE or all values in TUMOR_SEQ_ALLELE2.\n"
+            "maf: Contains instances where values in REFERENCE_ALLELE match values in TUMOR_SEQ_ALLELE2. "
+            "This is invalid. Please correct.\n",
         ),
     ],
+    ids=[
+        "matching_tsa1_tsa2",
+        "matching_tsa1_ref",
+        "invalid_tsa1",
+        "identical_ref_tsa2",
+        "identical_ref_tsa2_missing_tsa1",
+        "valid_ref_tsa2_missing_tsa1",
+        "missing_tsa2_ref",
+        "invalid_tsa1_identical_ref_tsa2",
+    ],
```

Comment:
Could you add a test case where we have NAs just to make sure the code doesn't break with NAs
- one test case of df with ref and tsa2 and tsa1 columns with NAs sprinkled in

Reply:
Sure. On it.